Makefile

```diff
@@ -504,6 +505,7 @@ mod-check: ## Check the go mod is clean and tidy.
 
 check-protos: ## Check the protobuf files are up to date.
 check-protos: clean-protos protos
+	@./tools/apply-expected-diffs.sh $(PROTO_GO_EXPECTED_DIFFS)
```

Comment:
should this go into `make protos` too? otherwise if I regenerate them my CI will fail

Reply:
Good idea.
Even though if you modify the `.proto` files and regenerate the `.pb.go` files then it's not very likely that the expected diffs can still be applied without changes to them, this might still make sense because at least it will give you an easy way to check what the `.pb.go` files with the expected diffs applied will look like.

Reply:
i didn't realize the diff won't work (unless the generated diff end up being further down in the file 🤷 ). But still not bad to do this automatically